app/src/test/java/com/codesoom/assignment/controllers/TaskControllerTest.java

```diff
@@ -0,0 +1,370 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.hamcrest.Matchers.hasSize;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.patch;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@DisplayName("Task Controller 클래스")
+class TaskControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private TaskService taskService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    private static final String TEST_TITLE = "test";
+    private static final String TEST_POSTFIX = "_!!!";
+    private static final Long VALID_ID = 1L;
+    private static final Long INVALID_ID = 0L;
+
+    @Nested
+    @DisplayName("GET Mapping - Task 조회 요청은")
+    class Describe_get_mapping {
+        @Nested
+        @DisplayName("등록된 Task가 있다면")
+        class Context_has_task {
+            final int givenTaskCnt = 5;
+
+            @BeforeEach
+            void prepare() {
+                List<Task> tasks = new ArrayList<>();
+                for (int i = 0; i < givenTaskCnt; i++) {
+                    tasks.add(getTask());
+                }
+
+                given(taskService.getTasks()).willReturn(tasks);
+            }
+
+            @Test
+            @DisplayName("200(Ok)과 등록된 Task 전체 리스트를 응답합니다.")
+            void it_return_ok_and_tasks() throws Exception {
+                mockMvc.perform(get("/tasks"))
+                        .andExpect(status().isOk())
+                        .andExpect(jsonPath("$", hasSize(givenTaskCnt)))
+                        .andDo(print());
+
+                verify(taskService, atLeast(1)).getTasks();
+            }
```

Comment:
해설 강의를 보고 verify 를 작성하였는데요.
```suggestion
                verify(taskService, atLeast(1)).getTasks();
```
단위테스트를 각각 할 때는 문제가 없는 한번에 할 때 다른 Context 클래스에서도 getTasks() 를 호출했던 것이 누적되어 1번 이상이 되게 되는데요. verify 이렇게 테스트 메서드마다 각각 써주는 것이 아닌가요?ㅜ

Reply:
`verify`에 집어넣은 `taskService`의 `getTasks`가 몇 번 호출되었는지를 보기 때문입니다. 메소드마다는 상관이 없어요. 이렇게 추측을 토대로 사용하시면 안됩니다. verify 소스코드를 보거나 관련 "공식" 문서를 꼭 찾아보고 사용하세요.

한편 `verify`는 이 테스트코드에서 필요하지 않아 보입니다. 이미 `hasSize`를 통해 결과를 확인하고 있기 때문이에요. 사용하신 의도는 무엇인가요? 저는 이 테스트에서 `verify`를 삭제하는 것을 권해드리고 싶어요. 오히려 좋은 테스트를 망치고 있습니다.

Reply:
네! verify 도 사용 목적이 분명해야되군요ㅜ   해설 강의를 보고 단순히 service 메소드의 서비스 실행 여부까지 체크해주는 건가 싶었습니다. 확인 감사합니다!!